de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/budgetissue/BudgetIssueType.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * #%L
+ * de.metas.serviceprovider.base
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.serviceprovider.budgetissue;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.stream.Stream;
+
+@AllArgsConstructor
+@Getter
+public enum BudgetIssueType
+{
+	INTERNAL("Internal"),
+	EXTERNAL("External");
+
+	private final String value;
+
+	public static BudgetIssueType getTypeByValue(final String value)
+	{
+		return Stream.of(values())
+				.filter(v -> v.getValue().equalsIgnoreCase(value))
+				.findFirst().orElse(null);
```

Comment:
> .filter(v -> v.getValue().equalsIgnoreCase(value))

why `equalsIgnoreCase`?  ...i would avoid it. Go with `equals`.

> .orElse(null);

Consider:
* returning optional
* or throw an exception